extensions/lifx/src/index.tsx

```diff
@@ -0,0 +1,375 @@
+import { ActionPanel, Color, Cache, List, Action, showToast, Toast, getPreferenceValues, Icon } from "@raycast/api";
+import { getProgressIcon } from "@raycast/utils";
+import { useState, useEffect } from "react";
+import axios from "axios";
+import { Api, Lights } from "./lib/interfaces";
+import { getHueIcon, getKelvinIcon, getLightIcon, parseDate } from "./lib/colorAlgos";
+import constants, { COLORS, effects } from "./lib/constants";
+import { cleanLights, SetEffect, SetLightState, toggleLight, FetchLights, checkApiKey } from "./lib/api";
+const hexToHsl = require("hex-to-hsl");
+
+export default function Command() {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [data, setData] = useState<Lights.Light[]>([]);
+  const [sideBar, setSideBar] = useState<boolean>(false);
+  const cache = new Cache();
+  const preferences = getPreferenceValues();
+
+  const config = {
+    headers: {
+      Authorization: "Bearer " + preferences.lifx_token,
+    },
+    timeout: 7000,
+  };
+
+  async function fetchLights() {
+    try {
+      if (!cache.has("lights")) {
+        const isTokenValid = await checkApiKey();
+        if (!isTokenValid) {
+          cache.set("lifx_token", JSON.stringify({ valid: true }));
+          await showToast({
+            style: Toast.Style.Failure,
+            title: "Invalid Token",
```

Comment:
```suggestion
            title: "Invalid Token",
            message: "Please check your token and try again",
```
Maybe something like this for error toasts?

Reply:
Yeah, I am going to add check token to all the functions interacting with the LIFX API and the set effect command.